posts/2024-03-21-Rust-1.77.0.md

```diff
@@ -0,0 +1,93 @@
+title: "Announcing Rust 1.77.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.77.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, you can get 1.77.0 with:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.77.0](https://doc.rust-lang.org/nightly/releases.html#version-77-2024-03-21).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.77.0 stable
+
+This release is relatively minor, but as always, even incremental improvements lead to a greater whole. A few of those changes are highlighted in this post, and others may yet fill more niche needs.
+
+### C-string literals
+
+Rust now supports C-string literals (`c"abc"`) which expand to a nul-byte
+terminated string in memory of type `&CStr`. This makes it easier to write code
+taking nul-terminated strings, with all of the relevant error checking (e.g.,
+lack of interior nul byte) implemented within the compiler.
+
+### `offset_of!`
+
+1.77.0 stabilizes [`offset_of!`] for struct fields, which provides access to the
+byte offset of the relevant public field of a struct. This macro is most useful
+when the offset of a field is required without an existing instance of a type.
+Implementing such a macro is already possible on stable, but without an
+instance of the type the implementation would require tricky unsafe code which
+makes it easy to get into undefined behavior.
+
+Users can now access the offset of a public field with `offset_of!(StructName,
+field)`. This expands to a `usize` expression with the offset in bytes from the
+start of the struct.
+
+Note that the output of the macro is not typically stable across compiler
+versions or across platforms. See the macro's documentation for details on when the output can be relied on.
+
+[`offset_of!`]: https://doc.rust-lang.org/nightly/std/mem/macro.offset_of.html
+
+### Enable strip in release profiles by default
+
+Cargo [profiles](https://doc.rust-lang.org/stable/cargo/reference/profiles.html)
+which do not enable [debug info](https://doc.rust-lang.org/stable/cargo/reference/profiles.html#debug) in
+outputs (e.g., `debug = 0`) will enable `strip = "debuginfo"` by default.
+
+This is primarily needed because the (precompiled) standard library ships with
+debug info, which means that statically linked results would include the
+debuginfo from the standard library even if the local compilations didn't
+explicitly request debuginfo.
```

Comment:
This paragraph uses both "debug info" and "debuginfo". I think it'd be better to choose one spelling and stick with it (through all three paragraphs, except for the TOML quote of course).